devito/propagator.py

```diff
@@ -178,8 +212,17 @@ def generate_space_loops_blocking(self, loop_body):
         ivdep = True
         remainder = False
         orig_loop_body = loop_body
+
+        inner_most_dim_passed = False  # used in oder to avoid at inner most dimension
+
         for spc_var, block_size in reversed(zip(list(self.space_dims), self.block_sizes)):
             dim_var = str(self._var_map[spc_var])
+
+            if self.auto_tune and inner_most_dim_passed:  # change block size into var
```

Comment:
Does @pvelesko have a citation or theory to back it up? In that case note it down as a comment in the code.  @mlange05 @mloubout @ggorman